AEPAnalytics/Sources/AnalyticsHitProcessor.swift

```diff
@@ -164,14 +165,24 @@ class AnalyticsHitProcessor: HitProcessing {
             // retry this hit later
             Log.warning(label: LOG_TAG, "\(#function) - Retrying Analytics hit, request with url \(url.absoluteString) failed with error \(connection.error?.localizedDescription ?? "") and recoverable status code \(connection.responseCode ?? -1)")
             completion(false)
-        } else if connection.responseCode == nil {
-            // retry this hit later if connection response code is nil (no network connectivity)
-            Log.warning(label: LOG_TAG, "\(#function) - Retrying Analytics hit, there is currently no network connectivity")
-            completion(false)
         } else {
-            // unrecoverable error. delete the hit from the database and continue
-            Log.warning(label: LOG_TAG, "\(#function) - Dropping Analytics hit, request with url \(url.absoluteString) failed with error \(connection.error?.localizedDescription ?? "") and unrecoverable status code \(connection.responseCode ?? -1)")
-            completion(true)
+            // handle network transport error
+            if let urlError = connection.error as? URLError, urlError.isRecoverable {
```

Comment:
Previously we would retry when `connection.responseCode == nil`, which from my understanding would be from any connection error. Now, however, we will only retry from a small list of URLErrors. Is there a concern that after this change some previously recoverable errors will no longer be recoverable?

Reply:
Yes we covered this in the guild meeting and we will follow the same pattern of retrying only for limited set of errors in all our extensions.